### PR TITLE
Add more tests for VLQ, ZigZag corner cases

### LINKED ISSUE #12: Add more tests for VLQ, ZigZag corner cases
Since serialisers are at the core of consensus protocol we need the following:
1) 100% code coverage with tests (using property testing)
2) EXPLICIT test cases for ALL corner cases (explicit original values, expected bytes). 
     It will be also useful if test will print some human readable log, so that it can be used as a reference data
  - minimal, maximal values
  - maximal 1 byte value, minimal 2 bytes value, 
  - maximal 2 bytes value, minimal 3 bytes value
  - etc.
3) equivalence of putUShort, putUInt, putULont for all small values
4) The same for ZigZag.

### PR DESCRIPTION
**Pull request by greenhat**
Close #12

### CODE REVIEW

File: src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala [hunk 1]
```diff
@@ -268,4 +272,259 @@ trait VLQReaderWriterSpecification extends PropSpec
...
+      }
+      byteBufReader(expected).getShort() shouldBe v
+    }
+
+    roundtrip(Short.MinValue, bytesFromInts(0xFF, 0xFF, 0x03))
+    roundtrip(-8193, bytesFromInts(0x81, 0x80, 0x01))
+    roundtrip(-8192, bytesFromInts(0xFF, 0x7F))
+    roundtrip(-65, bytesFromInts(0x81, 0x01))
+    roundtrip(-64, bytesFromInts(0x7F))
+    roundtrip(0, bytesFromInts(0))
```
Comment (aslesarenko) [inline src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala:375]: suggest to add the following cases: -8194, -8191, -66, -63, -1, 1, 62, 65, 8190, 8193
Also needed test cases for which show what happens when (0x81, 0x00) is read (and the same for all other whole in the byte arrays space)

File: src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala [hunk 2]
```diff
@@ -268,4 +272,259 @@ trait VLQReaderWriterSpecification extends PropSpec
...
+    def roundtrip(v: Int, expected: Array[Byte]): Unit = {
+      val bytes = byteArrayWriter().putUShort(v).toBytes
+      withClue(s"for value $v got bytes ${prettyPrint(bytes)} (expected ${prettyPrint(expected)}): \n") {
+        bytes shouldEqual expected
+      }
+      byteBufReader(expected).getUShort() shouldBe v
+    }
+
+    an[IllegalArgumentException] should be thrownBy roundtrip(-1, bytesFromInts(0))
+    roundtrip(0, bytesFromInts(0))
```
Comment (aslesarenko) [inline src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala:401]: please add at least one neighbour for each boundary case.

**Review by aslesarenko — CHANGES_REQUESTED** [review]
Looks quite good.

File: src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala (hunk 2 above)
Comment (greenhat) [reply src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala:401]: Done.

File: src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala (hunk 1 above)
Comment (greenhat) [reply src/test/scala/scorex/util/serialization/VLQReaderWriterSpecification.scala:375]: Done.

**Review by aslesarenko — APPROVED** [review]

**Review by kushti — APPROVED** [review]
